packages/itwinui-react/src/core/Tooltip/Tooltip.tsx

```diff
@@ -4,58 +4,250 @@
+      autoUpdate(referenceEl, floatingEl, update, {
+        animationFrame: updateOptions.animationFrame,
+        ancestorScroll: updateOptions.ancestorScroll,
+        ancestorResize: updateOptions.ancestorResize,
+        elementResize: updateOptions.elementResize,
+      }),
+    middleware: [
+      middleware.offset !== undefined ? offset(middleware.offset) : offset(4),
+      middleware.flip && flip(),
+      middleware.shift && shift(),
+      middleware.size && size(),
+      middleware.autoPlacement && autoPlacement(),
+      middleware.inline && inline(),
+      middleware.hide && hide(),
+    ].filter(Boolean),
+  });
+
+  const context = data.context;
+
+  const hover = useHover(context, {
+    enabled: controlledOpen == null,
+    delay: {
+      open: 50,
+      close: 250,
+    },
+    handleClose: safePolygon({ buffer: -Infinity }),
+  });
+
+  const focus = useFocus(context, {
+    enabled: controlledOpen == null,
+  });
+
+  const click = useClick(context, {
+    enabled: controlledOpen == null,
+  });
+
+  const dismiss = useDismiss(context, {
+    enabled: controlledOpen == null,
+  });
+
+  const role = useRole(context, { role: 'tooltip' });
+
+  const interactions = useInteractions([click, hover, focus, dismiss, role]);
+
+  return React.useMemo(
+    () => ({
+      open,
+      setUncontrolledOpen,
+      ...interactions,
+      ...data,
+    }),
+    [open, interactions, data],
+  );
+};
 
 /**
  * Basic tooltip component to display informative content when an element is hovered or focused.
- * Uses the {@link Popover} component, which is a wrapper around [tippy.js](https://atomiks.github.io/tippyjs).
+ * Uses [FloatingUI](https://floating-ui.com/).
  * @example
- * <Tooltip content='tooltip text' placement='top'><div>Hover here</div></Tooltip>
- * @example
- * const buttonRef = React.useRef();
- * ...
- * <Button ref={buttonRef} />
- * <Tooltip content='tooltip text' reference={buttonRef} />
+ * <Tooltip content='tooltip text' placement='top'>Hover here</Tooltip>
  */
-export const Tooltip = (props: TooltipProps) => {
-  const { content, children, className, style, visible, ref, id, ...rest } =
-    props;
+export const Tooltip = React.forwardRef((props, forwardRef) => {
+  const {
+    content,
+    children,
+    portal = true,
+    placement = 'top',
+    updateOptions,
+    middleware,
+    style,
+    className,
+    visible,
+    setReference,
+    ...rest
+  } = props;
+  const tooltip = useTooltip({ placement, visible, updateOptions, middleware });
+  const context = useGlobals();
 
-  return (
-    <Popover
-      visible={visible}
-      interactive={false}
-      content={
-        <Box
-          className={cx('iui-tooltip', className)}
-          style={style}
-          role='tooltip'
-          id={id}
-        >
-          {content}
-        </Box>
-      }
-      offset={[0, 4]}
-      ref={ref}
+  React.useEffect(() => {
+    setReference && setReference(tooltip.refs.setReference);
```

Comment:
should we instead ask for a `reference` and set it ourselves?

e.g.
```js
const { reference, ... } = props;


React.useEffect(() => {
  if (reference) {
    tooltip.refs.setReference(reference);
  }
}, [reference]);
```

Reply:
Updated

Reply:
update all leftover jsdocs to say `reference` instead of `setReference`

Reply:
and why not include correct deps in useEffect instead of ignoring eslint?
![](https://github.com/iTwin/iTwinUI/assets/9084735/d6a082c8-13a4-49c4-aa7d-5063f85f25dd)

similarly on line 128, `any` can be removed.
![](https://github.com/iTwin/iTwinUI/assets/9084735/0bccc597-81bb-417e-aaa6-41c93aee4d28)

Reply:
Updated